compiler/src/dotty/tools/dotc/transform/CapturedVars.scala

```diff
@@ -138,17 +142,21 @@ class CapturedVars extends MiniPhase with IdentityDenotTransformer { thisPhase =
     *  Also: If the ref type lhs is followed by a cast (can be an artifact of nested translation),
     *  drop the cast.
     */
-  override def transformAssign(tree: Assign)(using Context): Tree = {
+  override def transformAssign(tree: Assign)(using Context): Tree =
```

Comment:
Could yo update the comment above this method to describe the special case where we assign null to the temporary?